src/cst/parse.js

```diff
@@ -3,14 +3,22 @@
 import Document from './Document'
 import ParseContext from './ParseContext'
 
-export default function parse(src) {
+export default function parse(src, options) {
+  const defaultOptions = { computeLineOffsets: false }
+  options = Object.assign({}, defaultOptions, options)
   const cr = []
+  const lf = [0]
   if (src.indexOf('\r') !== -1) {
     src = src.replace(/\r\n?/g, (match, offset) => {
       if (match.length > 1) cr.push(offset)
       return '\n'
     })
   }
+  if (options.computeLineOffsets)
+    src = src.replace(/\n/g, (match, offset) => {
+      lf.push(offset + 1)
+      return '\n'
+    })
```

Comment:
`src.replace()` is far too heavy an operation for this; use [`src.indexOf('\n', fromIndex)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf) instead. There's no reason to mutate the string here.

Reply:
Cool. Will amend in the next revision.